apps/studio/state/recent-items.ts

```diff
@@ -0,0 +1,167 @@
+// Import necessary functions from valtio for state management
+import { proxy, subscribe } from 'valtio'
+// Import Tab and TabType types for defining recent items
+import { Tab, TabType } from './tabs'
+
+// Define the key for storing recent items in localStorage
+const RECENT_ITEMS_KEY = 'supabase_recent_items'
+// Define the maximum number of recent items to store
+const MAX_RECENT_ITEMS = 8
+
+const getStorageKey = (ref: string) => `${RECENT_ITEMS_KEY}_${ref}`
+
+// Define the structure of a recent item
+export interface RecentItem {
+  id: string // Unique identifier for the item
+  type: TabType // Type of the tab (e.g., sql, table)
+  label: string // Label for the item
+  timestamp: number // Timestamp of when the item was added
+  metadata?: {
+    // Optional metadata associated with the item
+    schema?: string // Schema name (if applicable)
+    name?: string // Name of the item (if applicable)
+    tableId?: number // Table ID (if applicable)
+    sqlId?: string // SQL ID (if applicable)
+  }
+}
+
+// Define the structure of the recent items state
+interface RecentItems {
+  items: RecentItem[] // Each project just has an array of items
+}
+
+interface RecentItemsStore {
+  [key: string]: RecentItems // Removed | undefined since we always initialize it
+}
+
+const defaultState: RecentItems = {
+  items: [],
+}
+
+export const recentItemsStore = proxy<RecentItemsStore>({})
```

Comment:
I'm not sure if it's the recent-items store or the tabs store, but if I open some tabs in one project, then navigate to another, I get this error:

<img width="671" alt="Screenshot 2025-03-20 at 18 33 14" src="https://github.com/user-attachments/assets/72e96044-3df5-47a8-a7ff-573423e3b8f7" />

We probably just need to scope this store to the project better. Here's an example: https://github.com/supabase/supabase/blob/ad831e8854eb56876d7207364513a6672ab74492/apps/studio/state/table-editor-table.tsx (from https://github.com/supabase/supabase/pull/34153).